client/src/modules/TeamDistribution/components/TeamDistributionCard/CardTitle.tsx

```diff
@@ -0,0 +1,52 @@
+import { Col, Row, Space, Tag, Typography } from 'antd';
+import { TeamOutlined, ClockCircleOutlined } from '@ant-design/icons';
+import { TeamDistributionDto, TeamDistributionDtoRegistrationStatusEnum } from 'api';
+import { DistributionPeriod } from './DistributionPeriod';
+
+type Props = {
+  distribution: TeamDistributionDto;
+};
+
+const { Text, Title } = Typography;
+
+export function CardTitle({ distribution }: Props) {
+  const renderTag = () => {
+    switch (distribution.registrationStatus) {
+      case TeamDistributionDtoRegistrationStatusEnum.Distributed:
+        return (
+          <Tag icon={<ClockCircleOutlined />} color="green">
+            distributed
+          </Tag>
+        );
+      case TeamDistributionDtoRegistrationStatusEnum.Completed:
+        return <Tag icon={<ClockCircleOutlined />}>without team</Tag>;
+      default:
+        return null;
+    }
+  };
+
+  return (
+    <Row gutter={24} wrap justify="space-between">
+      <Col>
+        <Title level={5}>{distribution.name}</Title>
+      </Col>
+      <Col>
+        <Space size="middle" wrap>
+          {distribution.registrationStatus === TeamDistributionDtoRegistrationStatusEnum.Completed ||
+          distribution.registrationStatus === TeamDistributionDtoRegistrationStatusEnum.Distributed
+            ? renderTag()
+            : distribution.minTotalScore !== 0 && (
+                <Text style={{ fontSize: 14 }} type="secondary">
+                  Min score {distribution.minTotalScore}
+                </Text>
+              )}
+          <Text style={{ fontSize: 14 }} type="secondary">
+            <TeamOutlined style={{ marginRight: 8 }} />
+            {distribution.strictTeamSize} members
+          </Text>
+          <DistributionPeriod startDate={distribution.startDate} endDate={distribution.endDate} />
+        </Space>
+      </Col>
+    </Row>
+  );
+}
```

Comment:
As for me — there's too many logic in render, I would suggest to extract some components here :)  
```suggestion
export function CardTitle({ distribution }: Props) {
  const isRegistrationCompletedOrDistributed =
    distribution.registrationStatus === TeamDistributionDtoRegistrationStatusEnum.Completed ||
    distribution.registrationStatus === TeamDistributionDtoRegistrationStatusEnum.Distributed;

  return (
    <Row gutter={24} wrap justify="space-between">
      <Col>
        <Title level={5}>{distribution.name}</Title>
      </Col>
      <Col>
        <Space size="middle" wrap>
          {isRegistrationCompletedOrDistributed ? (
            <RegistrationStatus status={distribution.registrationStatus} />
          ) : (
            <MinTotalScore score={distribution.minTotalScore} />
          )}
          <Text style={{ fontSize: 14 }} type="secondary">
            <TeamOutlined style={{ marginRight: 8 }} />
            {distribution.strictTeamSize} members
          </Text>
          <DistributionPeriod startDate={distribution.startDate} endDate={distribution.endDate} />
        </Space>
      </Col>
    </Row>
  );
}

type RegistrationStatusProps = {
  status: TeamDistributionDtoRegistrationStatusEnum;
};

const RegistrationStatus = ({ status }: RegistrationStatusProps) => {
  switch (status) {
    case TeamDistributionDtoRegistrationStatusEnum.Distributed:
      return (
        <Tag icon={<ClockCircleOutlined />} color="green">
          distributed
        </Tag>
      );
    case TeamDistributionDtoRegistrationStatusEnum.Completed:
      return <Tag icon={<ClockCircleOutlined />}>without team</Tag>;
    default:
      return null;
  }
};

type MinScoreProps = {
  score: number;
};

const MinTotalScore = ({ score }: MinScoreProps) => {
  if (!score) return null;

  return (
    <Text style={{ fontSize: 14 }} type="secondary">
      Min score {score}
    </Text>
  );
};
```

Reply:
thanks, i put it in the renderers file